src/oge/validation.py

```diff
@@ -52,64 +52,98 @@ def check_allocated_gf_matches_input_gf(year, gen_fuel_allocated):
     from the total input generation or fuel.
     """
     gf = load_data.load_pudl_table("denorm_generation_fuel_combined_eia923", year)
-    plant_total_gf = gf.groupby("plant_id_eia")[
+    plant_total_gf = gf.groupby(["plant_id_eia", "energy_source_code"])[
         [
             "net_generation_mwh",
             "fuel_consumed_mmbtu",
             "fuel_consumed_for_electricity_mmbtu",
         ]
     ].sum()
-    plant_total_alloc = gen_fuel_allocated.groupby("plant_id_eia")[
+
+    plant_total_alloc = gen_fuel_allocated.groupby(
+        ["plant_id_eia", "energy_source_code"]
```

Comment:
I'm wondering if in these groupbys we should include the `dropna=False` argument in case there is a missing ESC?

Reply:
Added. It does not change the outcome